lib/core/src/connection_pool.cpp

```diff
@@ -169,23 +169,32 @@ namespace irods
 
         if (_options.refresh_connections_when_resource_changes_detected) {
             std::uint64_t latest_mtime = 0;
+            std::uint16_t latest_mtime_millis = 0;
+            std::int32_t resc_count = 0;
 
             // Capture the latest time any resource was modified.
             // This will be used to track when a connection should be refreshed.
             // This helps with long-running agents.
-            for (auto&& row : irods::query{conn_ctxs_[0].conn.get(), "select max(RESC_MODIFY_TIME)"}) {
+
+            constexpr const char* gql = "select max(RESC_MODIFY_TIME), max(RESC_MODIFY_TIME_MILLIS), count(RESC_ID)";
```

Comment:
what is the thinking here?   this will get the largest millis in the database... is that sufficient/good/desired?

`max(RESC_MODIFY_TIME_MILLIS)`

Reply:
i guess we're only detecting if anything moved... so... max is fine...

Reply:
Correct. We can't concat anything using GenQuery and I'm trying to keep the number of operations down to a minimum.

I have another idea that may not require the need to track the number of resources at all.

Will let you know how it goes.

Reply:
gotcha.